src/utils/validateTypes.js

```diff
@@ -61,15 +61,15 @@ function isIfStatement(value) {
  * @param {unknown} value
  * @returns {value is variable}
  */
-const isDollarVar = value =>
-  (value.length > 0 && value[0] === "$") || value.includes(".$");
+const isDollarVar = value => value.match(/\$[A-Za-z0-9_-]+/g) !== null;
```

Comment:
I have to say that I like the non-regex version, both for simplicity, and possible perf reasons.

I have a feeling if we want to switch to a regex, then we should also add a proper unit tests for this function.

The question is, how much do we want to be manually parsing the variables from the value?

 I would prefer keeping the simpler version, but just making sure that the length is at least two chars: 

```js
const isDollarVar = value =>
  (value.length > 1 && value[0] === "$") || value.includes(".$");
```